models/gpr.py

```diff
@@ -0,0 +1,84 @@
+from abc import ABC, abstractmethod
+from sklearn.metrics import mean_squared_error
+from sklearn.gaussian_process.kernels import RBF, ConstantKernel, Matern
+from sklearn.gaussian_process import GaussianProcessRegressor
+import pandas as pd
+import numpy as np
+import matplotlib as mpl 
+from matplotlib import pyplot as plt
+from models.model import Model
+class GPR(Model):
+	def __init__(self, dist_array, user_name, dep_name, length, file, lat_n, lon_n):
+		self.dist = dist_array#array-like
+		self.user = file[user_name].values#array-like
+		self.dep = file[dep_name].values#array-like
+		self.depth = []
+		for i in range(len(self.dep)):
```

Comment:
Yeah this only doesn't work because Chris told me to invert the depth values so that on the graph so that we can actually see what it would look like underwater, so I have to modify the values in the process. Not inverting depth would let me do this and if another file had the depth as negative then it would switch to positive, so it might be better that way. However, I don't really have much knowledge of how expensive that for loop is, so depending on that, I might just be able to leave it as is, or it could be necessary to change.

Reply:
Since I changed it so that I don't invert the depth data until right before graphing, I switched it to this. Thanks!